src/xngin/stats/bandit_analysis.py

```diff
@@ -70,30 +71,41 @@ def _analyze_normal_binary(
         covariance: The posterior covariance matrix of the arm.
         context_link_functions: The link function to use.
         context: Optional context vector.
+        num_samples: Number of samples to draw for estimation.
         random_state: Use a fixed int for deterministic behavior in tests.
     """
+    # First derive the Credible Interval in latent space
+    if context is None:
+        latent_mean = float(mu[0])
+        latent_var = float(covariance[0, 0])
+    else:
+        latent_mean = float(context @ mu)
+        latent_var = float(context @ covariance @ context)
+
+    # Calculate 95% CI bounds
+    latent_stderr = np.sqrt(latent_var)
+    latent_ci_upper = latent_mean + 1.96 * latent_stderr
+    latent_ci_lower = latent_mean - 1.96 * latent_stderr
+    # Transform back into to probability space
+    latent_bounds = np.array([latent_ci_lower, latent_ci_upper])
+    ci_lower, ci_upper = context_link_functions(latent_bounds).tolist()
+
+    # Estimate the mean via MC integration given the non-linear link function.
+    # First sample in latent space
     rng = np.random.default_rng(random_state)
-    num_samples = 10000  # TODO: Make this configurable
-
     samples = rng.multivariate_normal(mean=mu, cov=covariance, size=num_samples)
     if context is not None:
         parameter_samples = samples @ context
     else:
         parameter_samples = samples
-    transformed_parameter_samples = context_link_functions(parameter_samples)
-    transformed_parameter_mean = transformed_parameter_samples.mean()
-    transformed_parameter_std = transformed_parameter_samples.std()
-    outcome_samples = rng.binomial(n=1, p=transformed_parameter_samples)
-
-    mean = outcome_samples.mean()
-    std = outcome_samples.std()
-    ci_upper = context_link_functions(
-        transformed_parameter_mean + 1.96 * transformed_parameter_std / np.sqrt(num_samples)
-    )
-    ci_lower = context_link_functions(
-        transformed_parameter_mean - 1.96 * transformed_parameter_std / np.sqrt(num_samples)
-    )
-    return float(mean), float(std), float(ci_upper), float(ci_lower)
+    # Convert to probabilities to compute the posterior predictive mean
+    prob_samples = context_link_functions(parameter_samples)
+    mean_prob = float(prob_samples.mean())
+
+    # Analytical standard deviation for bernoulli outcomes
+    std_dev = np.sqrt(mean_prob * (1 - mean_prob))
```

Comment:
What are we using std_dev for downstream?

Reply:
It's used in the FE plot; it's shown in the popup, to complement the CIs on the mean estimate.
But your question made me look more closely at the other analysis functions for different prior-likelihood pairs. I believe I found another CI calc issue in _analyze_normal. Can you look over my follow-up commit https://github.com/agency-fund/evidential-be/pull/119/commits/aafabaee26e56847f21f94ecf915b66f35c6c10d? 🙏